db/schema_tables.cc

```diff
@@ -1289,19 +1289,19 @@ static future<> merge_tables_and_views(distributed<service::storage_proxy>& prox
         return vp;
     });
 
-    co_await proxy.local().get_db().invoke_on_all([&] (replica::database& db) -> future<> {
-        // First drop views and *only then* the tables, if interleaved it can lead
-        // to a mv not finding its schema when snapshoting since the main table
-        // was already dropped (see https://github.com/scylladb/scylla/issues/5614)
-        co_await coroutine::parallel_for_each(views_diff.dropped, [&] (schema_diff::dropped_schema& dt) -> future<> {
-            auto& s = *dt.schema.get();
-            co_await db.drop_column_family(s.ks_name(), s.cf_name(), [&] { return dt.jp.value(); });
-        });
-        co_await coroutine::parallel_for_each(tables_diff.dropped, [&] (schema_diff::dropped_schema& dt) -> future<> {
-            auto& s = *dt.schema.get();
-            co_await db.drop_column_family(s.ks_name(), s.cf_name(), [&] { return dt.jp.value(); });
-        });
+    // First drop views and *only then* the tables, if interleaved it can lead
+    // to a mv not finding its schema when snapshoting since the main table
+    // was already dropped (see https://github.com/scylladb/scylla/issues/5614)
+    co_await coroutine::parallel_for_each(views_diff.dropped, [&db = proxy.local().get_db().local()] (schema_diff::dropped_schema& dt) {
+        auto& s = *dt.schema.get();
+        return db.drop_table_on_all(s.ks_name(), s.cf_name(), [&] { return dt.jp.value(); });
+    });
+    co_await coroutine::parallel_for_each(tables_diff.dropped, [&db = proxy.local().get_db().local()] (schema_diff::dropped_schema& dt) -> future<> {
+        auto& s = *dt.schema.get();
+        return db.drop_table_on_all(s.ks_name(), s.cf_name(), [&] { return dt.jp.value(); });
```

Comment:
Should we make database::drop_column_family private?

Reply:
It is, in the next patch.